src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1243,15 +1243,46 @@
         combined-list (vec (concat base-ext-list csr-ext-list))]
     (ensure-ext-list-has-cn-san subject combined-list)))
 
+(defn report-successful-cert-signing
+  "Log message and report to the activity service if available about successful cert signing."
+  [report-activity-or-nil message subject certname ip-address]
+  (let [commit {:service {:id "puppetserver"}
+                :subject {:id subject
+                          :name subject
+                          :type "users"}
+                :objects [{:type "node" :id certname :name certname}]
+                :events [{:type "sign-certificate"
+                          :what "node"
+                          :description (str "certificate_successfully_signed")
+                          :message message}]
+                :ip_address ip-address}]
+    (log/info message)
+    (if report-activity-or-nil
+      (report-activity-or-nil {:commit commit}))))
+
+(defn generate-cert-message-from-request
```

Comment:
This function name is a little misleading since it doesn't _just_ generate the message but also does the reporting. We should probably either just have this return message. Maybe have it return the message, ip, subject, and certname and rename it? Or just rename the function as it is to something else.